internal/controller/postgrescluster/patroni_test.go

```diff
@@ -516,41 +524,51 @@ func TestReconcilePatroniSwitchover(t *testing.T) {
 			status  string
 			soType  string
 			target  string
-			check   func(*testing.T, error)
+			check   func(*testing.T, error, *v1beta1.PostgresCluster)
 		}{
 			{
 				desc:    "Switchover not enabled",
 				enabled: false,
-				check: func(t *testing.T, err error) {
+				check: func(t *testing.T, err error, cluster *v1beta1.PostgresCluster) {
 					assert.NilError(t, err)
+					assert.Equal(t, cluster.Status.Patroni.SwitchoverTimeline, nil32)
+					assert.Equal(t, cluster.Status.Patroni.Switchover, nilString)
```

Comment:
🤔 I don't think `assert.Equal` shows the values behind non-nil pointers. When I [try it](https://go.dev/play/p/N1rXlmK9Wfc) I see memory locations instead:

```
assertion failed: 0xc000188300 (*string) != <nil> (*string)
assertion failed: 0xc000188300 (*string) != 0xc000188310 (*string)
```

⛏️ `{address} != {address}` doesn't help me understand what happened or implement a fix. These nil checks can be simpler with the nil constant.

```go
assert.Assert(t, cluster.Status.Patroni.Switchover == nil)
// assertion failed: cluster.Status.Patroni.Switchover is not nil
```

📝 I forget what property(s) of Go allows comparison to nil like this. Nil is untyped. Pointers are comparable. Nil is assignable to any pointer.